pymc4/model/base.py

```diff
@@ -38,6 +38,7 @@ def __init__(self, name=None, graph=None, session=None, **config):
             session = tf.Session(graph=graph)
         self.session = session
         self.observe(**config)
+        self.temp_graph = tf.Graph()
```

Comment:
I would use this pattern for creating and destroying the temporary graph.  I'm using a pymc3 model here because I think the pattern is similar, but let me know if it isn't!  Note that you will only need to implement the equivalent of the `temp_model` method below.

```python
import pymc3 as pm


def check_in_model_context():
    try:
        m = pm.modelcontext(None)
        return f'In model context! {len(m.vars)} vars!'
    except TypeError:
        return 'Not in model context!'

class Model(object):
    def __init__(self):
        self._temp_model = None
        
    @contextmanager
    def temp_model(self):
        self._temp_model = pm.Model()
        try:
            with self._temp_model:
                yield
        finally:
            self._temp_model = None
    
    def calculation(self):
        print(check_in_model_context())
        with self.temp_model():
            pm.Normal('x', 0, 1)
            print(check_in_model_context())
            
        print(check_in_model_context())
        with self.temp_model():
            print(check_in_model_context())
        print(check_in_model_context())
```

Reply:
decorator for inclass function is causing problems.. I will look into